src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,476 @@
+        response-handler (make-reponse-handler socket)
+        pending (atom nil)
+        zero-bytes (python/bytes "" "UTF-8")
+        client-info (py->lisp (.getsockname socket))
+        client* (atom {:*1 nil :*2 nil :*3 nil :*e nil ;; keeps track of the latest evaulation results.
+                       :eval-ns nil ;; the last eval ns
+                       })]
+    (try
+      (info "Connection accepted" :info client-info)
+      ;; Need to load the `clojure.core` alias because cider uses it
+      ;; to test for availability of features.
+      (eval (read-string "(ns user (:require clojure.core))"))
+      (swap! client* assoc :eval-ns *ns*)
+      (loop [data (.recv socket recv-buffer-size)]
+        (if (= data zero-bytes)
+          (do (info :socket-closing client-info)
+              (.close socket))
+          (let [data (.decode data "UTF-8")
+                data (if-let [p @pending]
+                       (let [s (str p data)]
+                         (reset! pending nil)
+                         s)
+                       data)
+                [requests unprocessed] (bc/decode-all data {:keywordize-keys true})]
+            (debug :requests requests)
+            (when (not (str/blank? unprocessed))
+              (reset! pending unprocessed))
+            (doseq [request requests]
+              (try
+                (handler (assoc request :client* client*) response-handler)
+                (catch python/Exception e
+                  (error :request-handler-unexpected-exception (pr-str e)))))
+            (recur (.recv socket recv-buffer-size)))))
+      (catch python/Exception e
+        (error :client-connection-error :client client-info :exception e)
+        (error (traceback/format_exc))))))
+
+(defn server-make [opts]
+  "Create and return a `socketserver/TCPServer` serving nREPL clients
+  according to ``opts` at 127.0.0.1 at a random available port.
+
+  See `ops` for the operation supported by the server.
+
+  The nREPL starts at the `user` namespace and binds `*1`, `*2`, `*3`
+  and `*e` to the ultimate, penultimate, antepenultimate evaluation
+  result and last exception message respectively.
+
+  ``opts`` is a map of options with the following optional keys
+
+  :host The host address to bind to, defaults to 127.0.0.1.
+
+  :port The port number to listen to, defaults to 0 which means to
+  pickup a random available port.
+
+  See :lpy:fn:`on-connect` for additionally supported ``opts`` keys.
+
+  Known limitations:
+
+  1. All client connections share the same environment at the moment,
+  which is the env that the server runs in. This could change in the
+  future to isolate the clients interactions from each other.
+
+  2. The session uuids are ignored and only created to satisfy the
+  initial clone op."
+  (let [{:keys [host port] :or {host "127.0.0.1" port 0}} opts
+        handler (python/type (name (gensym "mytcphandler")) (python/tuple [socketserver/StreamRequestHandler])
```

Comment:
```suggestion
        handler (python/type (name (gensym "NReplTCPHandler")) (python/tuple [socketserver/StreamRequestHandler])
```

Might be a slightly nicer and more diagnostic name.

Reply:
Updated.